### PR TITLE
Add r-rcppdate package

### PR DESCRIPTION
**Pull request by Shelnutt2**
This is a new dependency for the already existing [r-nanotime package](https://github.com/conda-forge/r-nanotime-feedstock).
Upstream r-nanotime added RcppDate as a dependency in 0.3.0.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by ihnorton** [comment]
[x] confirming they are willing to be listed there (@ihnorton)

**Comment by Shelnutt2** [comment]
@conda-forge/help-r this is ready for review. The two linter errors I believe are false positives. The package was created from the CRAN repo with the https://github.com/bgruening/conda_r_skeleton_helper , as such the license SPDX is not lining up. Happy to adjust the identifier if needed.

Secondly this PR does not touch the example recipe, [it's still there](https://github.com/conda-forge/staged-recipes/blob/54cc62f6f434262fc3b20d54095cb457f0841f8e/recipes/example/meta.yaml).

File: recipes/r-rcppdate/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - r-base
+
+test:
+  commands:
+    - $R -e "library('RcppDate')"           # [not win]
+    - "\"%R%\" -e \"library('RcppDate')\""  # [win]
+
+about:
+  home: https://github.com/eddelbuettel/rcppdate
+  license: GPL-2
```
Comment (dbast) [inline recipes/r-rcppdate/meta.yaml:38]:
```suggestion
  license: GPL-2.0-or-later
```
GPL (>= 2) -> https://spdx.org/licenses/GPL-2.0-or-later.html

**Comment by cbrueffer** [comment]
Please add `conda-forge/r` as maintainer.

File: recipes/r-rcppdate/meta.yaml (hunk 1 above)
Comment (Shelnutt2) [reply recipes/r-rcppdate/meta.yaml]: Done, thanks for the feedback.

**Comment by Shelnutt2** [comment]
> Please add `conda-forge/r` as maintainer.

I have added `conda-forge/r` to the maintainers list.

**Comment by dbast** [comment]
@Shelnutt2 Thanks!

**Pull request merged**